File: resources/capability-bundle/.claude/skills/privacy-provenance/scripts/scan-output.py
import json
import re
import sys


payload = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
text = str(payload.get("text", ""))
patterns = [
    ("private-key", re.compile(r"-----BEGIN (?:RSA |EC |OPENSSH )?PRIVATE KEY-----")),
    ("api-key-like", re.compile(r"\bsk-[A-Za-z0-9_-]{16,}\b")),
    ("windows-user-path", re.compile(r"[A-Za-z]:\\Users\\[^\\\s]+")),
    ("unix-user-path", re.compile(r"/(?:Users|home)/[^/\s]+")),
]

findings = [{"name": name} for name, pattern in patterns if pattern.search(text)]
sys.stdout.write(json.dumps({"safe": not findings, "findings": findings}, separators=(",", ":")))

File: resources/capability-bundle/.claude/skills/privacy-provenance/scripts/scan-output.mjs
const payload = JSON.parse(process.argv[2] ?? '{}');
const text = String(payload.text ?? '');

const patterns = [
  { name: 'private-key', pattern: /-----BEGIN (?:RSA |EC |OPENSSH )?PRIVATE KEY-----/u },
  { name: 'api-key-like', pattern: /\bsk-[A-Za-z0-9_-]{16,}\b/u },
  { name: 'windows-user-path', pattern: /[A-Za-z]:\\Users\\[^\\\s]+/u },
  { name: 'unix-user-path', pattern: /\/(?:Users|home)\/[^/\s]+/u },
];

const findings = patterns
  .filter(({ pattern }) => pattern.test(text))
  .map(({ name }) => ({ name }));

process.stdout.write(JSON.stringify({ safe: findings.length === 0, findings }));
